nixos/modules/services/networking/unbound.nix

```diff
@@ -6,41 +6,116 @@ let
+           '';
+        };
+  in {
+    options = {
+      server = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          server: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      remote-control = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          remote-control: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      stub-zone = mkOption {
+        type = with types; listOf settingsType;
+        default = [ ];
+        description = ''
+          stub-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      forward-zone = mkOption {
+        type = with types; listOf settingsType;
+        default = [ ];
+        example = literalExample ''
+          [
+            {
+              name = ".";
+              forward-addr = "1.1.1.1@853#cloudflare-dns.com";
+            }
+            {
+              name = "example.org.";
+              forward-addr = [
+                "1.1.1.1@853#cloudflare-dns.com"
+                "1.0.0.1@853#cloudflare-dns.com"
+              ];
+            }
+          ];
+        '';
+        description = ''
+          forward-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      python = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          python: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+    };
+  };
+
+  yesOrNo = v: if v then "yes" else "no";
+
+  toOption = n: v: "${toString n}: ${v}";
+
+  toConf = n: v:
+    if builtins.isFloat v then (toOption n (builtins.toJSON v))
+    else if isInt v       then (toOption n (toString v))
+    else if isBool v      then (toOption n (yesOrNo v))
+    else if isString v    then (toOption n v)
+    else if isList v      then (concatMapStringsSep "\n  " (toConf n) v)
+    else throw (traceSeq v "services.unbound.settings: : unexpected type");
+
+  clauseConf = clause: conf: if conf == { } then "" else
+    (concatStringsSep "\n  " (
+      ["${clause}:"] ++ (
+        mapAttrsToList toConf (
+          filterAttrs (_: v: v != null) conf
```

Comment:
You filter out `null`'s, but the type doesn't allow nulls to begin with. So you can either remove this filter or adjust the type to allow `null`.